src/components/Modal/Modal.js

```diff
@@ -2,49 +2,55 @@ import AddCarForm from '../AddCarForm/AddCarForm';
 import Typography from '../../../styles/Typography';
 import StyledButton from '../../../styles/StyledButton';
 import StyledFlex from '../../../styles/StyledFlex';
+import useStore from '../../hooks/useStore';
 
-function AddCarModal(props) {
-	if (!props.show) {
-		return null;
-	}
-	return (
-		<StyledFlex
-			justifyContent="center"
-			alignItems="center"
-			position="fixed"
-			top="0"
-			right="0"
-			bottom="0"
-			left="0"
-			background="var(--transparent)"
-			onClick={props.onClose}
-		>
+function Modal() {
+	const toggleModalState = useStore(state => state.toggleModalState);
+	const modalState = useStore(state => state.modalState);
+
+	if (modalState === true) {
```

Comment:
Same here with the early exit. Also, I guess the `modalState` can always be either `true` or `false` so in this case you could just do `if(modalState)` instead.

Reply:
adjusted accordning to feedback.